docs/docker.md

```diff
@@ -0,0 +1,108 @@
+# Intro
+This document explains how to use our different Dockerfiles and some technical details about them.
+
+## Intro to bin scripts
+Our scripts in `bin` directory have been updated so we can run them with docker. To do this, you have to set the `DOCKER_ENABLED=true` env variable. This can be done using the .env file, or exporting it in the terminal.
+
+## Intro to Docker
+`Dockerfile.dev` - This Dockerfile is specific for local development purposes. It's optimized to get the live reloading working and assumes the resulting image size is not a concern here.
+
+`Dockerfile` - This Dockerfile is optimized to run in production environments. It ensures a minimal image size to run the app, by installing the minimal needed dependencies, and also adds Jemalloc to improve the Ruby memory management.
+
+## Intro to Docker Compose
+`docker-compose.yml` - This compose file is intented to use only for local development, NOT FOR PRODUCTION. It runs all the services we need in order to run our Rails server, and can be easily extended to include more services if your app needs to. It depends on the `Dockerfile.dev` file and has some "watch" mechanisms to build or reload the app when files change.
+
+`docker-compose.test.yml` - This compose file is only used for the test environment since it includes a standalone chrome-server service we need to run our E2E tests.
+
+# Usage
+To build and run the project in development we recommend using Docker Compose instead of standalone Docker, as all necessary services, networks and configurations are already included in the docker compose file. For ease of use, you can also run the scripts provided directly in your terminal.
+
+## bin scripts
+
+### How to run tests
+```bash
+./bin/rspec
+```
+> [!WARNING]
+> By default this command will not stop the db and chrome-server services when the process completes. To achieve this you can run `./bin/rspec --rm` instead
+
+### How to use Rails console
+```bash
+./bin/rails c
+```
+
+### How to start the Rails server
+Currently there is no script for doing this with docker. Check how to start the Rails server using Docker Compose in the next section.
```

Comment:
but running `./bin/rails s` will use Docker if I have the ENV VAR and Docker running because of https://github.com/rootstrap/rails_api_base/pull/732/files#diff-f7968f0a612c299c9774edbd1a3c2169dbc5a1809ba9208ccc50beef597364b3 right?

Reply:
Yes, that's correct. The issue is that the resulting command will be
```bash
docker compose run web ./bin/rails s
```

There is a difference between doing a `run` and doing `up`.
When you use `up` all the services are executed and all the ports are exported. If you use `run <service>` instead, the service is started (in this case the web service) along with all its dependencies (servicies specified in the `depends_on` attribute) but the ports are never linked. So you are going to see you server started, but hitting the port `3000` on your browser will just no work.

Reply:
Actually I was just reading the [run command docs](https://docs.docker.com/reference/cli/docker/compose/run/#description). Looks like we can use the `--service-ports` flag to actually export them... Maybe it would be nice to test it and see if there are no collisions. cc: @PerezIgnacio @smerlo @axelpontetto